examples/gno.land/r/demo/gnofundme/gnofundme.gno

```diff
@@ -0,0 +1,319 @@
+		return renderCampaign(parts[1])
+	}
+	return "unknown page"
+}
+
+func renderHomePage() string {
+	balance = banker.GetCoins(std.CurrentRealm().Addr()).AmountOf("ugnot")
+	// define empty buffer
+	var b bytes.Buffer
+
+	// users counter
+	userc := 1
+	b.WriteString("# Welcome to GnoFundMe\n\n")
+	b.WriteString(ufmt.Sprintf("Balance: %d ugnot.\n", balance))
+
+	b.WriteString("# Campaigns :\n\n")
+	campaigns.Iterate("", "", func(key string, value interface{}) bool {
+		c := value.(*GFMpkg.Campaign)
+		campaignURL := ufmt.Sprintf("%scampaign/%s", campaignPrefix, key)
+		b.WriteString(
+			ufmt.Sprintf("## [Campaign %s](%s)\n", key, campaignURL),
+		)
+		b.WriteString(
+			ufmt.Sprintf(
+				"### Description: %s\n", c.Description,
+			),
+		)
+
+		return false
+	})
+	b.WriteString("# Users:\n")
+
+	users.Iterate("", "", func(key string, value interface{}) bool {
+		u := value.(*GFMpkg.User)
+		b.WriteString(
+			ufmt.Sprintf(
+				"## User #%d: %s, Key: %s\n", userc, u.GetName(), key,
+			),
+		)
+		b.WriteString(
+			ufmt.Sprintf(
+				"- email: %s\n - address: %s\n", u.GetEmail(), u.GetAddress(),
+			),
+		)
+		userc++
+		return false
+	})
+	return b.String()
+}
+
+func renderCampaign(id string) string {
+	// get the campaign if it exists
+	campaign, ok := campaigns.Get(id)
+	if !ok {
+		return "campaign not found"
+	}
+
+	// define an empty buffer
+	var b bytes.Buffer
+	c := campaign.(*GFMpkg.Campaign)
+	b.WriteString(
+		ufmt.Sprintf("# Campaign %s\n", id),
+	)
+	b.WriteString(
+		ufmt.Sprintf("## Title: %s\n", c.Title),
+	)
+	b.WriteString(
+		ufmt.Sprintf("## Description: %s\n", c.Description),
+	)
+	b.WriteString(
+		ufmt.Sprintf("### Goal: %d GNOT", c.Goal),
+	)
+	b.WriteString(
+		ufmt.Sprintf("********** Current: %d GNOT\n", c.Current/1_000_000),
+	)
+	b.WriteString(
+		ufmt.Sprintf("### Begin: %s", c.Begin.String()),
+	)
+	b.WriteString(
+		ufmt.Sprintf("  ------------->  Deadline: %s\n", c.Deadline.String()),
+	)
+	b.WriteString(
+		ufmt.Sprintf("### Owner: %s, %s\n", c.Owner.GetName(), c.Owner.GetAddress()),
+	)
+	b.WriteString(
+		ufmt.Sprintf("### Beneficiary: %s, %s\n", c.Beneficiary.GetName(), c.Beneficiary.GetAddress()),
+	)
+	b.WriteString(
+		ufmt.Sprintf("## Contributors: %d\n", len(c.GetContributors())),
+	)
+	if len(c.GetContributors()) != 0 {
+		contributors := c.GetContributors()
+		for _, contributor := range contributors {
+			if contributor.GetDisplay() {
+				b.WriteString(
+					ufmt.Sprintf(
+						"### %s: %dugnot, %s\n", contributor.GetUser().GetName(), contributor.Amount, contributor.Date.String(),
+					), // added the date of contribution in the display, but adds "m=+1726419880.000000001" after the date =>> the problems comes from time.now(), had to modify and use time.Unix(time.Now().Unix(), 0) to solve the problem.
+				)
+			} else {
+				b.WriteString(
+					ufmt.Sprintf(
+						"### Anonymous: %dugnot\n", contributor.Amount,
+					),
+				)
+			}
+		}
+	}
+	return b.String()
+}
```

Comment:
What if there is a million contributors? I would limit this to the latest 5 - 10, with the option to fetch more somehow.